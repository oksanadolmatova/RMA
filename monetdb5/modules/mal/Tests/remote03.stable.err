stderr of test 'remote03` in directory 'monetdb5/modules/mal` itself:


# 04:04:13 >  
# 04:04:13 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32121" "--set" "mapi_usock=/var/tmp/mtest-3168/.s.monetdb.32121" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/default//Linux/var/MonetDB/mTests_monetdb5_modules_mal"
# 04:04:13 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/default//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 32121
# cmdline opt 	mapi_usock = /var/tmp/mtest-3168/.s.monetdb.32121
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/default//Linux/var/MonetDB/mTests_monetdb5_modules_mal
# cmdline opt 	gdk_debug = 536870922

# 04:04:13 >  
# 04:04:13 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "--host=/var/tmp/mtest-3168" "--port=32121"
# 04:04:13 >  

MAPI  = (monetdb) /var/tmp/mtest-3168/.s.monetdb.32121
QUERY = # let connect figure out itself how to connect to the running db
        uri := sabaoth.getLocalConnectionURI();
        conn:str := remote.connect(uri, "monetdb", "monetdb");
        
        # this is invalid, as io.print is an invalid module, and bla will be
        # taken as function
        function tst(conn:str);
        res:str := remote.put(conn,:void);
        res := remote.exec(conn, "io.print", "bla");
        catch SyntaxException:str;
        	io.print("Received first error from the remote site");
        	io.printf("!%s\n", SyntaxException);
        exit SyntaxException;
        
        # now do it with a function that returns something (a BAT in this case)
        bn:= bat.new(:str);
        b := remote.put(conn,bn);
        catch MALException:str;
        	io.print("Received second init error from the remote site");
        	io.printf("!%s\n", MALException);
        exit MALException;
        io.print(b);
        b := remote.exec(conn, "inspect", "getAtomNames");
        catch MALException:str;
        	io.print("Received second error from the remote site");
        	io.printf("!%s\n", MALException);
        exit MALException;
        io.print(b);
        
        c:bat[:str] := remote.get(conn, b);
        io.print(c);
        catch MALException:str;
        	io.print("Received third error from the remote site");
        	io.printf("!%s\n", MALException);
        exit MALException;
        
        # io.print only outputs things, but returns nothing
        ah:str := remote.put(conn,:void);
        io.print(ah);
        ah:str := remote.exec(conn, "io", "print", b);
        catch MALException:str;
        	io.print("Received fourth error from the remote site");
        	io.printf("!%s\n", MALException);
        exit MALException;
        
        d:void := remote.get(conn, ah);
        io.print(d);
        catch MALException:str;
        	io.print("Received fifth error from the remote site");
        	io.printf("!%s\n", MALException);
        exit MALException;
        end tst;
        tst(conn);
        
        catch SyntaxException:str;
        	io.print("Received an error from the remote site");
        	io.printf("!%s\n", SyntaxException);
        exit SyntaxException;
        # help testweb a bit, since currently no cleanup is done on server
        # shutdown
        remote.disconnect(conn);
        catch MALException:str;
        	io.print("Received an error from the remote site");
        exit MALException;
        
ERROR = !SyntaxException:remote.exec:(mapi:monetdb://monetdb@vienna.da.cwi.nl/mTests_monetdb5_modules_mal) rmt0_X_0_void := io.print.bla();

# 04:04:13 >  
# 04:04:13 >  "Done."
# 04:04:13 >  

