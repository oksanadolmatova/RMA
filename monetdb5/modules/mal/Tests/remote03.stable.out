stdout of test 'remote03` in directory 'monetdb5/modules/mal` itself:


# 04:04:13 >  
# 04:04:13 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32121" "--set" "mapi_usock=/var/tmp/mtest-3168/.s.monetdb.32121" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/default//Linux/var/MonetDB/mTests_monetdb5_modules_mal"
# 04:04:13 >  

# MonetDB 5 server v11.24.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2016 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:32121/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-3168/.s.monetdb.32121
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 04:04:13 >  
# 04:04:13 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "--host=/var/tmp/mtest-3168" "--port=32121"
# 04:04:13 >  

[ "Received first error from the remote site"	]

# 04:04:13 >  
# 04:04:13 >  "Done."
# 04:04:13 >  

